Common: Extend Battery Status with Remaining Time and Low Extent State

MaEtUgR opened the pull request:
Allow the autopilot to optionally publish its:
- estimation of the remaining time in seconds the battery can still be used before it's empty.
This information can be showed to the user or used by an external algorithm to further plan the usage of the battery/vehicle.
- low battery state telling the user or external components to what extent the battery is empty.
This information can be used to update the UI with appropriate warnings and inform external components to trigger energy saving or other risk minimizing behaviour.

FYI @julianoes @dogmaphobic 

**EDIT:** fixes #809

julianoes commented on message_definitions/v1.0/common.xml:
```diff
@@ -2474,6 +2474,27 @@
         <description>Payload battery</description>
       </entry>
     </enum>
+    <enum name="MAV_BATTERY_LOW_STATE">
```
I would call it `BATTERY_CHARGE_STATE` because `LOW_STATE` implies that it's already low.

MaEtUgR replied:
Your reasoning makes sense, seems I was too burried in my own naming scheme.

julianoes commented on message_definitions/v1.0/common.xml:
```diff
@@ -2474,6 +2474,27 @@
+      </entry>
+      <entry value="5" name="MAV_BATTERY_LOW_STATE_FAILED">
+        <description>Battry failed, damage unavoidable</description>
+      </entry>
```
For smart batteries the state `UNHEALTHY` could also be good which prevents taking off with a "broken" battery.

MaEtUgR replied:
Yes or at least lets the user know he needs to order a new one.

MaEtUgR replied:
I extended the definition to errors which prevent usage in general.

MaEtUgR commented:
I rebased on master and addressed the comment see commit message:
> Thanks to a review comment by @WickedShell I learned that extensions
only available in MAVLink 2 get filled with zeroes in older MAVLink
versions. Hence 0 has to be the value meaning no useful data is provided.
It's a small scrifice to have 1 second battery usage remaining defined
as the value for a completely empty battery.

Thanks for mentioning @WickedShell!
